lib/rspec/support/method_signature_verifier.rb

```diff
@@ -85,6 +85,10 @@ def has_kw_args_in?(args)
         # contain keyword arguments?
         def could_contain_kw_args?(args)
           return false if args.count <= min_non_kw_args
+          if args.count <= max_non_kw_args && Hash === args.last
```

Comment:
The reason I've added `Hash === args.last` is that we use `could_contain_kw_args?` in https://github.com/rspec/rspec-support/blob/master/lib/rspec/support/method_signature_verifier.rb#L383 and I'm guarding against the matcher.

Reply:
Fair enough, this could be inlined into one if though?